server/src/main/java/com/cloud/api/dispatch/ParamProcessWorker.java

```diff
@@ -258,11 +258,11 @@ public void processParameters(final BaseCmd cmd, final Map params) {
                 }
 
             } catch (final IllegalArgumentException e) {
-                s_logger.error("Error initializing command " + cmd.getCommandName() + ", field " + field.getName() + " is not accessible.");
+                s_logger.error("Error initializing command " + cmd.getCommandName() + ", field " + field.getName() + " is not accessible.", e);
```

Comment:
Why do you need to log the stack trace twice? I mean, the exception is re-thrown as a cloudruntime. Wherever it is catch, it will probably be logged.

In the Jira ticket you mention that the problem (confusion) is caused by the same message in the stack trace. Why not improve the messaging then, instead of logging printing the strack trace twice?

Reply:
Thanks for your reply. The exception is thrown by a new exception cloudruntime, but the original exception variable "e" is lost when throwing the new exception, so the logs will not show if the exception type is IllegalArgumentException or IllegalAccessException.

It's true that it will be better to improve the log message, but I'm not so sure if I can change the message correctly. So it's a safe way to add original stack trace information here, at least we can know the true exception types from the logs generated without any harm.

Reply:
@rafaelweingartner are you satisfied with this answer?

Reply:
Well, that was an answer. I would prefer to fix/improve the log messaging than to log twice the same stack trace. What do you think?

Reply:
I do not see that point, sorry. An unexpected error is caught and logged (not rethrown) and than a more generic message/exception is thrown. Other than unifying the two catch clauses in a catch (IllegalArgumentException|IllegalAccessException e), I don't see the harm here. We could improve by not logging at all and adding the original exception as a root cause to the CloudRuntimeException. Is that what you mean?

Reply:
That is what I am trying to say :)

Reply:
ok, so there is a misunderstanding here. The original exception is not rethrown, but a new one @lzh3636 . The question is can you add it to the throw and delete the error log instead?

Reply:
Yes, thank you, it's done :)